### PR TITLE
Use rust-toolchain instead of RustConfig

### PR DESCRIPTION
**Pull request by anp**
Something pretty similar to this is currently powering rusty-dash.com after I moved it over to Heroku last night/this morning. I think it should be backwards compatible with RustConfig's VERSION setup as well.

Fixes https://github.com/emk/heroku-buildpack-rust/issues/11

### CODE REVIEW

**Comment by emk** [comment]
This looks great! Unfortunately, I'm in meetings for the next couple of days and won't have time to test this myself, so I might need one more favor before merging this.

> I think it should be backwards compatible with RustConfig's VERSION setup as well.

Could you please test the backwards-compatibility support, and confirm that it still works? I mean, obviously the code looks correct, but I don't want to risk people in the field by merging a cool patch without testing it. :-)

**Comment by dikaiosune** [comment]
Makes sense! (reminds me how nice it would be for Heroku to let you pin a custom buildpack to a commit/tag!)

In my fork, I created a `RustConfig` file with `VERSION=1.20.0` as the contents, and ran `./test_buildpack`:

```
 ~/c/heroku-buildpack-rust   master  heroku-rust-cargo-hello  nano RustConfig
 ~/c/heroku-buildpack-rust   master  heroku-rust-cargo-hello  cd ..
 ~/c/heroku-buildpack-rust   master  ./test_buildpack                     Wed Nov 15 23:55:54 2017
WARNING: The GROUPS variable is not set. Defaulting to a blank string.
Starting herokubuildpackrust_testbuild_1 ...
Starting herokubuildpackrust_testbuild_1 ... done
Attaching to herokubuildpackrust_testbuild_1
testbuild_1  | -----> Checking for new releases of Rust 1.20.0 channel
testbuild_1  | info: checking for self-updates
testbuild_1  | info: syncing channel updates for '1.20.0-x86_64-unknown-linux-gnu'
testbuild_1  | info: latest update on 2017-08-31, rust version 1.20.0 (f3d6973f4 2017-08-27)
```

Looks like the correct setting got picked up, since the latest stable on my machine (and what I've built with this container before) is 1.21.0.

File: bin/compile [hunk 1]
```diff
@@ -26,6 +26,11 @@ RUST_SKIP_BUILD=0
 # `BUILD_PATH` to the correct directory in the `RustConfig`
 BUILD_PATH=""
 
+# Load our toolchain configuration, if any was specified.
+if [ -f "$BUILD_DIR/rust-toolchain" ]; then
+    VERSION=$(cat $BUILD_DIR/rust-toolchain)
```
Comment (emk) [inline bin/compile]: I think it might be safe to quote this:

```sh
VERSION="$(cat "$BUILD_DIR/rust-toolchain")"
```

File: bin/detect [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-if [ -f "$1/RustConfig" ] || [ -f "$1/Cargo.toml" ]; then
+if [ -f "$1/RustConfig" ] || [ -f "$1/rust-toolchain" ] || [ -f "$1/Cargo.toml" ]; then
```
Comment (emk) [inline bin/detect]: I don't think it's necessary to add `rust-toolchain` here, but it shouldn't hurt anything, either. So we can keep it.

**Comment by dikaiosune** [comment]
Sorry, forgot to comment here after I pushed a new commit. Addressed the requested changes (I think).

**Comment by anp** [comment]
Ping! Hope you're well. Would be nice to unfork :).

**Comment by emk** [comment]
CC @Floppy

The only thing blocking this PR is a one-line fix [mentioned here](https://github.com/emk/heroku-buildpack-rust/pull/29#pullrequestreview-77387835). If anybody wants to fix it, I'd be happy to merge the change. (I'd do it myself, but I'm on deadline for work stuff.)

**Comment by anp** [comment]
Hey! Sorry about that, not sure what happened to those notifications. Just pushed a fix!

**Comment by emk** [comment]
Thank you! Merging. Let me know if you have any problems.

**Pull request merged**